core/src/main/java/org/apache/carbondata/core/datastore/blocklet/BlockletEncodedColumnPage.java

```diff
@@ -0,0 +1,187 @@
+  /**
+   * fallback future task queue;
+   */
+  private ArrayDeque<Future<FallbackEncodedColumnPage>> fallbackFutureQueue;
+
+  BlockletEncodedColumnPage(ExecutorService fallbackExecutorService,
+      EncodedColumnPage encodedColumnPage) {
+    this.encodedColumnPageList = new ArrayList<>();
+    this.fallbackExecutorService = fallbackExecutorService;
+    this.encodedColumnPageList.add(encodedColumnPage);
+    // if dimension page is local dictionary enabled and encoded with local dictionary
+    if (encodedColumnPage.isLocalDictionaryEnabled() && encodedColumnPage
+        .isLocalDictGeneratedPage()) {
+      this.isLocalDictEncoded = true;
+      // get first page dictionary
+      this.pageLevelDictionary = encodedColumnPage.getPageDictionary();
+    }
+  }
+
+  /**
+   * Below method will be used to add column page of a column
+   *
+   * @param encodedColumnPage
+   * encoded column page
+   * @throws ExecutionException
+   * failure in fallback
+   * @throws InterruptedException
+   * failure during fallback
+   */
+  void addEncodedColumnColumnPage(EncodedColumnPage encodedColumnPage)
+      throws ExecutionException, InterruptedException {
+    // if local dictionary is false or column is encoded with local dictionary then
+    // add a page
+    if (!isLocalDictEncoded || encodedColumnPage.isLocalDictGeneratedPage()) {
+      this.encodedColumnPageList.add(encodedColumnPage);
+      // merge page level dictionary values
+      if (null != this.pageLevelDictionary) {
+        pageLevelDictionary.mergerDictionaryValues(encodedColumnPage.getPageDictionary());
+      }
+    } else {
+      // if older pages were encoded with dictionary and new pages are without dictionary
+      isLocalDictEncoded = false;
+      pageLevelDictionary = null;
+      this.fallbackFutureQueue = new ArrayDeque<>();
+      LOGGER.info(
+          "Local dictionary Fallback is initiated for column: " + encodedColumnPageList.get(0)
+              .getActualPage().getColumnSpec().getFieldName());
+      // submit all the older pages encoded with dictionary for fallback
+      for (int pageIndex = 0; pageIndex < encodedColumnPageList.size(); pageIndex++) {
+        fallbackFutureQueue.add(fallbackExecutorService.submit(
+            new FallbackColumnPageEncoder(encodedColumnPageList.get(pageIndex), pageIndex)));
+      }
+      //add to page list
+      this.encodedColumnPageList.add(encodedColumnPage);
+    }
+  }
+
+  /**
+   * Return the list of encoded page list for a column in a blocklet
+   *
+   * @return list of encoded page list
+   */
+  public List<EncodedColumnPage> getEncodedColumnPageList() {
+    // if fallback queue is null then for some pages fallback was initiated
+    if (null != this.fallbackFutureQueue) {
+      try {
+        // check if queue is not empty
+        while (!fallbackFutureQueue.isEmpty()) {
+          // get the head element of queue
+          Future<FallbackEncodedColumnPage> fallbackTask = fallbackFutureQueue.getFirst();
```

Comment:
I think just use `poll` and `future.get` to fill the pages instead of using `sleep`